docs/bump-upstream.md

```diff
@@ -16,4 +16,6 @@
    - [`splunk-otel-dotnet-install.sh.template`](../script-templates/splunk-otel-dotnet-install.sh.template)
    - [`Splunk.OTel.DotNet.psm1.template`](../script-templates/Splunk.OTel.DotNet.psm1.template)
 
+1. Update [compatibility Matrix data](../tools/MatrixHelper).
```

Comment:
I think it's worth to describe (as sub-bullets) here, which (type of) metadata is necessary to manually recheck.
Since there seem to be many (nuget versions, metrics, etc), we can easily miss something.

Reply:
See, eaf78cb23416d97de76fc92efd63c78c4b594416

I hope I covered all important data